public/jira/raw/rollback/rollback.ts

```diff
@@ -0,0 +1,311 @@
+/**
+ * This module is responsible for rolling back Jira issues to their previous states based on their changelogs.
+ * It processes changes to fields such as Sprint, Fix Versions, Parent associations, and Status,
+ * ensuring that issues are accurately reverted to their historical configurations.
+ */
+import { parseDateISOString } from "../../../date-helpers";
+import {
+  Change,
+  FixVersion,
+  JiraIssue,
+  Sprint,
+  Status,
+} from "../../shared/types";
+
+interface RolledBackMetadata {
+  rolledbackTo: Date;
+  didNotExist?: boolean;
+}
+
+interface RolledBackJiraIssue extends JiraIssue {
+  rollbackMetadata: RolledBackMetadata;
+}
+
+interface FieldLookup<T> {
```

Comment:
nice! always interesting see `Map` being used :D

Reply:
I usually go for objects unless a `Map` I need to guarantee the order of the keys, just because I am more familiar with objects. But since a Map was already used here, I kept it as it is